File: dotnet/src/MyDotey.SCF.Simple/Source/StringProperty/PropertieFile/PropertiesFileConfigurationSourceConfig.cs
using System;

namespace MyDotey.SCF.Source.StringProperty.PropertiesFile
{
    /**
     * @author koqizhao
     *
     * May 17, 2018
     */
    public class PropertiesFileConfigurationSourceConfig : DefaultConfigurationSourceConfig
    {
        private String _fileName;

        protected PropertiesFileConfigurationSourceConfig()
        {

        }

        public String getFileName()
        {
            return _fileName;
        }

        public override String ToString()
        {
            return String.Format("{0} {{ name: {1}, fileName: {2} }}", GetType().Name, getName(), getFileName());
        }

        public new class Builder
            : DefaultConfigurationSourceConfig.DefaultAbstractBuilder<Builder, PropertiesFileConfigurationSourceConfig>
        {
            protected override PropertiesFileConfigurationSourceConfig newConfig()
            {
                return new PropertiesFileConfigurationSourceConfig();
            }

            public Builder setFileName(String fileName)
            {
                getConfig()._fileName = fileName;
                return this;
            }

            public override PropertiesFileConfigurationSourceConfig build()
            {
                if (string.IsNullOrWhiteSpace(getConfig().getFileName()))
                    throw new ArgumentNullException("fileName is null or empty");

                getConfig()._fileName = getConfig()._fileName.Trim();
                if (!getConfig()._fileName.EndsWith(".properties"))
                    getConfig()._fileName += ".properties";

                return base.build();
            }
        }
    }
}

File: java/scf-simple/src/main/java/org/mydotey/scf/source/stringproperty/propertiesfile/PropertiesFileConfigurationSourceConfig.java
package org.mydotey.scf.source.stringproperty.propertiesfile;

import org.mydotey.scf.DefaultConfigurationSourceConfig;

/**
 * @author koqizhao
 *
 * May 17, 2018
 */
public class PropertiesFileConfigurationSourceConfig extends DefaultConfigurationSourceConfig {

    private String _fileName;

    protected PropertiesFileConfigurationSourceConfig() {

    }

    public String getFileName() {
        return _fileName;
    }

    @Override
    public String toString() {
        return String.format("%s { name: %s, fileName: %s }", getClass().getSimpleName(), getName(), getFileName());
    }

    public static class Builder extends
            DefaultConfigurationSourceConfig.DefaultAbstractBuilder<Builder, PropertiesFileConfigurationSourceConfig> {

        @Override
        protected PropertiesFileConfigurationSourceConfig newConfig() {
            return new PropertiesFileConfigurationSourceConfig();
        }

        public Builder setFileName(String fileName) {
            getConfig()._fileName = fileName;
            return this;
        }

        @Override
        public PropertiesFileConfigurationSourceConfig build() {
            if (getConfig().getFileName() == null || getConfig().getFileName().trim().isEmpty())
                throw new IllegalArgumentException("fileName is null or empty");

            getConfig()._fileName = getConfig()._fileName.trim();
            if (!getConfig()._fileName.endsWith(".properties"))
                getConfig()._fileName += ".properties";

            return (PropertiesFileConfigurationSourceConfig) super.build();
        }

    }

}
